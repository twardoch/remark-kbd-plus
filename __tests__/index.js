import dedent from 'dedent'
import unified from 'unified'
import reParse from 'remark-parse'
import remarkStringify from 'remark-stringify'
import rehypeStringify from 'rehype-stringify'
import remark2rehype from 'remark-rehype'
import remarkCustomBlocks from '../../remark-custom-blocks'

import plugin from '../src/'


const render = text => unified()
  .use(reParse, {
    footnotes: true,
  })
  .use(remarkCustomBlocks, {
    secret: 'spoiler',
  })
  .use(plugin)
  .use(remark2rehype)
  .use(rehypeStringify)
  .processSync(text)

const fixture = dedent`
  Blabla ||ok|| kxcvj ||ok foo|| sdff

  sdf |||| df

  sfdgs | | dfg || dgsg | qs

  With two pipes: \||key|| you'll get ||key||.

  It can contain inline markdown:

  * ||hell[~~o~~](#he)?||

  It cannot contain blocks:

  * ||hello: [[secret]]?||
`


describe('parses kbd', () => {
  it('parses a big fixture', () => {
    const {contents} = render(fixture)
    expect(contents).toMatchSnapshot()
  })

  it('escapes the start marker', () => {
    const {contents} = render(dedent`
      ||one|| \||escaped|| ||three|| \|||four|| ||five||
    `)
    expect(contents).toContain('||escaped||')
    expect(contents).toContain('|<kbd>four</kbd>')
  })
})

test('to markdown', () => {
  const {contents} = unified()
    .use(reParse)
    .use(remarkStringify)
    .use(plugin)
    .processSync(fixture)

  expect(contents).toMatchSnapshot()
})
