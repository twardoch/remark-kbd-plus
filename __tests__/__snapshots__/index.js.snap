// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`parses kbd parses a big fixture 1`] = `
"<p>Blabla <kbd>ok</kbd> kxcvj <kbd>ok foo</kbd> sdff</p>
<p>sdf |||| df</p>
<p>sfdgs | | dfg || dgsg | qs</p>
<p>With two pipes: ||key|| you'll get <kbd>key</kbd>.</p>
<p>It can contain inline markdown:</p>
<ul>
<li><kbd>hell<a href=\\"#he\\"><del>o</del></a>?</kbd></li>
</ul>
<p>It cannot contain blocks:</p>
<ul>
<li><kbd>hello: [[secret]]?</kbd></li>
</ul>"
`;

exports[`to markdown 1`] = `
"Blabla ||ok|| kxcvj ||ok foo|| sdff

sdf |||| df

sfdgs | | dfg || dgsg | qs

With two pipes: \\\\||key|| you'll get ||key||.

It can contain inline markdown:

-   ||hell[~~o~~](#he)?||

It cannot contain blocks:

-   ||hello: \\\\[[secret]]?||
"
`;
